seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java

```diff
@@ -199,8 +202,33 @@ public Object computeForValue(Expression expression, Object[] inputFields) {
             return ((StringValue) expression).getValue();
         }
         if (expression instanceof Column) {
-            int idx = inputRowType.indexOf(((Column) expression).getColumnName());
-            return inputFields[idx];
+            Column columnExp = (Column) expression;
+            String columnName = columnExp.getColumnName();
+            int index = inputRowType.indexOf(columnName, false);
+            if (index != -1) {
+                return inputFields[index];
+            } else {
+                String fullyQualifiedName = columnExp.getFullyQualifiedName();
+                String[] columnNames = fullyQualifiedName.split("\\.");
+                int deep = columnNames.length;
+                SeaTunnelDataType parDataType = inputRowType;
+                SeaTunnelRow parRowValues = new SeaTunnelRow(inputFields);
+                Object res = parRowValues;
+                for (int i = 0; i < deep; i++) {
+                    if (parDataType instanceof MapType) {
+                        return ((Map) res).get(columnNames[i]);
+                    }
+                    parRowValues = (SeaTunnelRow) res;
+                    int idx = ((SeaTunnelRowType) parDataType).indexOf(columnNames[i], false);
+                    if (idx == -1) {
+                        throw new IllegalArgumentException(
+                                String.format("can't find field [%s]", fullyQualifiedName));
+                    }
+                    parDataType = ((SeaTunnelRowType) parDataType).getFieldType(idx);
+                    res = parRowValues.getFields()[idx];
```

Comment:
```
  public int indexOf(String fieldName, boolean throwExceptionWhenNotFound) {
        for (int i = 0; i < fieldNames.length; i++) {
            if (fieldNames[i].equals(fieldName)) {
                return i;
            }
        }
        if (throwExceptionWhenNotFound) {
            throw new IllegalArgumentException(String.format("can't find field [%s]", fieldName));
        } else {
            return -1;
        }
    }
``` Logical conflict？
cc  @hailin0 @Hisoka-X

Reply:
cc @Hisoka-X @hailin0

Reply:
check the field name is in this struct, if not choose throw exception or return -1 index let's the caller handle.

Reply:
<img width="957" alt="image" src="https://github.com/apache/seatunnel/assets/76689593/264f2763-0186-473e-b2b6-270aa928a6f6">
Repetitive code logic

Reply:
Thanks @zhilinli123 for ping me. I think it's different. `indexOf` only throw sub fieldname. The `fullyQualifiedName` and `columnNames[i]` not equals. Like field `user.age`, we should throw `user.age` not `age`.

Reply:
i agree with @Hisoka-X